scripts/install/install_macos.sh

```diff
@@ -566,22 +572,34 @@ install_package()
   decrease_indent
 }
 
-# create_manager_yml creates the manager.yml at the specified path, containing opamp information.
-create_manager_yml()
+create_supervisor_config()
 {
-  manager_yml_path="$1"
-  if [ ! -f "$manager_yml_path" ]; then
-    info "Creating manager yaml..."
+  supervisor_yml_path="$1"
+  if [ ! -f "$supervisor_yml_path" ]; then
+    info "Creating supervisor config..."
 
     # Note here: We create the file and change permissions of the file here BEFORE writing info to it
     # We do this because the file may contain a secret key, so we want 0 window when the
     # file is readable by anyone other than root
-    command printf '' >> "$manager_yml_path"
-    chmod 0600 "$manager_yml_path"
-
-    command printf 'endpoint: "%s"\n' "$OPAMP_ENDPOINT" > "$manager_yml_path"
-    [ -n "$OPAMP_LABELS" ] && command printf 'labels: "%s"\n' "$OPAMP_LABELS" >> "$manager_yml_path"
-    [ -n "$OPAMP_SECRET_KEY" ] && command printf 'secret_key: "%s"\n' "$OPAMP_SECRET_KEY" >> "$manager_yml_path"
+    command printf '' >> "$supervisor_yml_path"
+    chmod 0600 "$supervisor_yml_path"
+
+    command printf 'server:\n' > "$supervisor_yml_path"
+    command printf '  endpoint: "%s"\n' "$OPAMP_ENDPOINT" >> "$supervisor_yml_path"
+    command printf '  headers:\n' >> "$supervisor_yml_path"
+    [ -n "$OPAMP_SECRET_KEY" ] && command printf '    Authorization: "Secret-Key %s"\n' "$OPAMP_SECRET_KEY" >> "$supervisor_yml_path"
+    [ -n "$OPAMP_LABELS" ] && command printf '    X-Bindplane-Attribute: "%s"\n' "$OPAMP_LABELS" >> "$supervisor_yml_path"
```

Comment:
Better way to accomplish this might be to add a snippet like this:
```yaml
agent:
  description:
    non_identifying_attributes:
      service.labels: "$labels"
```

I think this would make the install_id work (haven't tested yet but assuming there's still trouble with it)

Reply:
Tried this but still no luck

Reply:
The labels are working for me, but only on the unreleased versions (v1.68.0+) -- this is due to the recent changes to labels we are making BindPlane side. So I think this is good where it is right now.